### PR TITLE
Admin UI console results visual improvements

### PR DESCRIPTION
**Pull request by Autophagy**

### CODE REVIEW

File: app/scripts/controllers/console.js [hunk 1]
```diff
@@ -79,6 +84,43 @@ angular.module('console', ['sql'])
           });
         };
 
+        $scope.isGeopoint = function isGeopoint(dataType) {
```
Comment (m-kharbat) [inline app/scripts/controllers/console.js]: All these checking functions can be moved to a separate service which will make them reusable and a lot easier to test

File: app/scripts/controllers/console.js [hunk 2]
```diff
@@ -232,5 +281,107 @@ angular.module('console', ['sql'])
...
+  .directive('formattedArray', function ($filter) {
+    return {
+      restrict: 'E',
+      scope: {
+        array: '=',
+        typesarray: '=',
+        expand: '='
+      },
+      templateUrl: 'views/formatted-array.html',
+      link: function($scope) {
```
Comment (m-kharbat) [inline app/scripts/controllers/console.js]: it's a bit confusing to have `$scope` injected in the link function, the standard way is to use `scope` https://www.bennadel.com/blog/2716-when-to-use-scope-vs-scope-in-angularjs.htm

File: app/scripts/controllers/console.js [hunk 3]
```diff
@@ -232,5 +281,107 @@ angular.module('console', ['sql'])
...
+          } else {
+            return $filter('columnTypeClass')($scope.typesarray[1]);
+          }
+        };
+
+        $scope.getArrayType = function getArrayType() {
+          return $scope.typesarray[0] == 101 ? 'Set' : 'Array';
+        };
+
+        $scope.isObject = function isObject(index) {
```
Comment (m-kharbat) [inline app/scripts/controllers/console.js]: these functions are defined twice.. you can put them in a service and reuse them in the directive, which will also be easier to test

File: app/scripts/controllers/console.js [hunk 4]
```diff
@@ -232,5 +281,107 @@ angular.module('console', ['sql'])
...
+  })
+  .directive('formattedObject', function () {
+    return {
+      restrict: 'E',
+      scope: {
+        object: '=',
+        expand: '='
+      },
+      templateUrl: 'views/formatted-object.html',
+      link: function($scope) {
```
Comment (m-kharbat) [inline app/scripts/controllers/console.js]: same comment about $scope in link function

File: app/styles/dark-theme-variables.less [hunk 5]
```diff
@@ -254,3 +255,12 @@
 
 @code-bg-color: @lightred;
 @code-color:    @darkred;
+
+//
+// Console Results Colourisation
+// ---------------------------------------------------
+
+@numeric-result-color: #a0a1ff;
```
Comment (m-kharbat) [inline app/styles/dark-theme-variables.less]: all the colors , appart from the grays :( are defined in the App Colors section with color names to make them reusable and easily identifiable, also so that we don't have too many shades of the same color 

https://github.com/crate/crate-admin/blob/master/app/styles/light-theme-variables.less#L33

File: app/styles/light-theme-variables.less [hunk 6]
```diff
@@ -255,3 +256,12 @@
 
 @code-bg-color:     @lightred;
 @code-color:        @darkred;
+
+//
+// Console Results Colourisation
+// ---------------------------------------------------
+
+@numeric-result-color: #181bd0;
```
Comment (m-kharbat) [inline app/styles/light-theme-variables.less]: same comment about the colors

File: app/tests/filter/numbers.js [hunk 7]
```diff
@@ -0,0 +1,122 @@
+describe('floor', function() {
```
Comment (m-kharbat) [inline app/tests/filter/numbers.js]: these tests are not related to the feature, you can add them here https://github.com/crate/crate-admin/pull/321/files or make a separate pr

File: app/views/formatted-array.html [hunk 8]
```diff
@@ -0,0 +1,13 @@
+<div class="formatted-array">
```
Comment (m-kharbat) [inline app/views/formatted-array.html]: maybe we can add a suffix to know which files are directive templates and which are views ?

File: app/views/formatted-array.html [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+  <div class="formatted-array__title"><a ng-click="toggleExpand()"><i class="fa fa-chevron-right" ng-class="isExpanded ? 'expanded-icon' : ''"></i> {{ getArrayType() }}<span class="formatted-array__length">[{{ getArrayLength() }}]</span></a></div>
+  <div class="formatted-array__elements" ng-if="isExpanded">
+    <div ng-repeat="element in array track by $index">
+      <span ng-if="isObject($index)" class="formatted-array__element">
+        <div class="key">{{ formatArrayindex($index) }}</div> <div class="seperator">:</div> <formatted-object object="element" expand="false"></formatted-object>
```
Comment (m-kharbat) [inline app/views/formatted-array.html]: i didn't see where some classes are defined example, key, value, separator etc

File: app/views/formatted-object.html [hunk 10]
```diff
@@ -0,0 +1,16 @@
+<div class="formatted-object">
+  <div class="formatted-object__title"><a ng-click="toggleExpand()"><i class="fa fa-chevron-right" ng-class="isExpanded ? 'expanded-icon' : ''"></i> Object</a></div>
```
Comment (m-kharbat) [inline app/views/formatted-object.html]: how do you feel about using http://fontawesome.io/icon/caret-right/ (because we also using in the nodes view )?

File: app/scripts/controllers/console.js [hunk 11]
```diff
@@ -232,5 +281,107 @@ angular.module('console', ['sql'])
       }
     };
   })
+  .directive('lazyLoadScroll', function ($window) {
+    return {
+      restrict: 'A',
+      link: function (scope, element, attrs) {
+        var e = element[0];
+        $($window).bind('scroll', function () {
```
Comment (m-kharbat) [inline app/scripts/controllers/console.js]: this looks a bit weird because you are injecting a $window to use as a jquery selector.. you can use `window` directly as a selector in jquery and you already have access to .scroll() event handler https://api.jquery.com/scroll/ it looks like .bind() is deprecated

File: app/views/formatted-array.html (hunk 8 above)
Comment (Autophagy) [inline app/views/formatted-array.html]: Similarly, do you think the formatted-array/object templates should be moved into their own folder? app/views/templates or app/templates or something like this?

**Comment by Autophagy** [comment]
@m-kharbat I think that's all your comments addressed apart from https://github.com/crate/crate-admin/pull/340#discussion_r92606960 - would be interesting to know your thoughts on it & whether it would involve refactoring other templates like https://github.com/crate/crate-admin/blob/mika/consoleResultsImprovements/app/views/editor.html

File: app/views/formatted-array.html (hunk 8 above)
Comment (m-kharbat) [inline app/views/formatted-array.html]: I would suggest to just add `-template` or `-directive`
I think changing the folder would require a lot of change in the grunt build
